Objects/setobject.c

```diff
@@ -2408,13 +2538,23 @@ PyTypeObject PyFrozenSet_Type = {
 PyObject *
 PySet_New(PyObject *iterable)
 {
-    return make_new_set(&PySet_Type, iterable);
+    PyObject *rv;
+
+    Py_BEGIN_CRITICAL_SECTION(iterable);
```

Comment:
We don't want a critical section here. In the cases where we need to lock `iterable`, we should make that decision deeper down when we know more about the type of `iterable`.